grr/server/grr_response_server/gui/ui/store/client_page_facade_test.ts

```diff
@@ -486,4 +486,65 @@ describe('ClientPageFacade', () => {
          done();
        });
      });
+
+  it('fetches client data only after selectedClient$ is subscribed to', fakeAsync(() => {
+    expect(httpApiService.fetchClient).not.toHaveBeenCalled();
+    clientPageFacade.selectedClient$.subscribe();
+
+    // This is needed since selected client is updated in a timer loop
+    // and the first call is scheduled after 0 milliseconds (meaning it
+    // will happen right after it was scheduled, but still asynchronously).
+    tick(1);
+    discardPeriodicTasks();
```

Comment:
Is the frequent use of `discardPeriodicTasks()` needed?

Reply:
It's a good question, unfortunately the docs regarding `discardPeriodicTasks()` weren't too generous. From playing a little with them, it seems like they are only required once. Will update the tests accordingly.